src/app/core/services/label.service.ts

```diff
@@ -73,33 +75,72 @@ export class LabelService {
    * Stores the json data from Github api into the list of arrays in this service
    * @param labels: the json data of the label
    */
-  private populateLabelLists(labels: Array<{}>): void {
-
-    for (const label of labels) {
-      // Get the name and color of each label and store them into the service's array list
-      const labelName = String(label['name']).split('.');
-      const labelType = labelName[0];
-      const labelValue = labelName[1];
-      const labelColor = String(label['color']);
-
-      // Check for duplicate labels
-      if (this.labelColorMap.has(labelValue)) {
-        continue;
-      }
+  private storeLabelData(labels: Array<{}>): void {
 
-      this.labelColorMap.set(labelValue, labelColor);
-      const labelList = this.getLabelList(labelType);
+    // Parse Input Data to Label[]
+    const labelData: Label[] = this.parseLabelData(labels);
+    const expectedLabels: Label[] = Label.getRequiredLabels();
 
-      if (labelList !== undefined) {
-        labelList.push({labelValue: labelValue, labelColor: labelColor});
+    expectedLabels.forEach(label => {
+      // Checks if remote data set already contains correct label.
+      if (labelData.filter(remoteLabel => remoteLabel.equals(label)).length !== 0) {
+        this.saveLabelData(label);
+        return;
+      }
+      // Finds for a label that has a valid name & type but invalid color.
+      const existingLabel: Label = labelData.find(remoteLabel => remoteLabel.getFormattedName() === label.getFormattedName());
+      if (existingLabel === undefined) {
+        // Create new Label (Could not find a label with the same name & category)
+        this.githubService.createLabel(label.getFormattedName(), label.labelColor);
+      } else {
+        // Update Label Color (Found a label with same name and category BUT it contains different color.)
+        this.githubService.updateLabel(label.getFormattedName(), label.labelColor);
```

Comment:
Alternative logic:
```
filter labels matching name only.
if results found == 0:
  create new label.
else if size == 1:
  update color if color mismatched.
else:
  assertion failure
```

Reply:
@damithc 
A query, Github currently does not allow the creation of 2 labels with the same name. Thus wouldn't there only be 2 cases? Finding a label with a matching name or none?

Reply:
That's why that branch is an assertion failure. If the execution reaches that branch, either there is a bug in the code or GitHub doesn't behave as expected.

Reply:
> That's why that branch is an assertion failure. If the execution reaches that branch, either there is a bug in the code or GitHub doesn't behave as expected.

Sure. Would you prefer that the application prevent login on such a case or try to correct the issue by deleting those labels from github and re-creating them.

Reply:
Will prevent login as discussed. 👍